scripts/PearlPowderISIS/pearl_cycle_factory.py

```diff
@@ -0,0 +1,155 @@
+# pylint: disable=anomalous-backslash-in-string, too-many-branches, superfluous-parens
+
+from mantid.simpleapi import *
+import pearl_routines
+
+
+def get_cycle_dir(number, currentdatadir):
```

Comment:
Why are you suppressing `too-many-branches`? the function `get_cycle_dir` has a really high cyclomatic complexity and should be replaced with an associative container.